apps/condo/domains/property/components/panels/Builder/MapConstructor.ts

```diff
@@ -567,10 +567,39 @@ class MapEdit extends MapView {
         this.notifyUpdater()
     }
 
+    private getNextUnit (id: string): BuildingUnit {
+        const units = this.map.sections.map(section => [...section.floors].reverse().map(floor => floor.units)).flat(2)
+        const unitIndex = units.findIndex(unit => unit.id === id)
+        const nextIndex = unitIndex + 1
+        return units[nextIndex] || null
+    }
+
+
+    private removeFloor (sectionIdx: number, floorIndex: number): void {
+        const floorToRemove = this.map.sections[sectionIdx].floors[floorIndex]
+        this.map.sections[sectionIdx].floors.splice(floorIndex, 1)
```

Comment:
is it possible to get undefined at sectionIdx?

Reply:
Put if condition in case someone calls this function outside remove unit